libs/utils/src/http/endpoint.rs

```diff
@@ -143,6 +143,31 @@ pub fn auth_middleware<B: hyper::body::HttpBody + Send + Sync + 'static>(
     })
 }
 
+pub fn add_static_response_header_middleware<B>(
+    header: &'static str,
+    value: &'static str,
+) -> Middleware<B, ApiError>
+where
+    B: hyper::body::HttpBody + Send + Sync + 'static,
+{
+    Middleware::post_with_info(move |mut response, request_info| async move {
+        let headers = response.headers_mut();
+        if headers.contains_key(header) {
+            tracing::warn!(
+                "{} response already contains header {}",
+                request_info.uri(),
+                header,
+            );
+        } else {
+            headers.insert(
+                header,
+                value.parse().expect("any UTF-8 string should work here"),
```

Comment:
If you'd parse the value outside of this `async move {}` in the body, the panic would be accessible to the `main` during start, instead of on first request.

If I remember correctly hyper would also prefer an already converted HeaderName, which could be converted in the main body.

Reply:
This seems to be implemented.